platform/lambda/lambda.go

```diff
@@ -663,11 +664,24 @@ func isCreatingRole(err error) bool {
 func toEnv(env config.Environment, stage string) *lambda.Environment {
 	m := aws.StringMap(env)
 	m["UP_STAGE"] = &stage
+	for k, v := range dotEnvVars() {
+		m[k] = &v
```

Comment:
will confirm tomorrwo but I think this is a bug, taking the address of the key or value, since it'll always give you the last value.

It'd be nice to move this out too, seems magical within a function to read a file. On a side note I think I'd like to have

```
Environment:  p.environment(stage),
```

instead of

```
Environment:  toEnv(p.config.Environment, stage),
```

I'd maybe just run `godotenv.Read()` inline within that method instead since `dotEnvVars()` only adds a log and doesn't tweak the logic.

Reply:
thanks for the comments.

And you are right about taking the address and its fixed now. Also tested with multiple env vars now.

Also made `toEnv` a method `environment` on Platform.

Let me know how it looks now!

Thanks!